File: dmesg.pl
#!/usr/bin/perl

use strict;
use warnings;

my @dmesg_new = ();
my $dmesg = "/bin/dmesg";
my @dmesg_old = `$dmesg`;
my $now = time();
my $uptime = `cat /proc/uptime | cut -d"." -f1`;
my $t_now = $now - $uptime;

foreach my $line ( @dmesg_old )
{
	chomp( $line );
	if( $line =~ m/\[\s*(\d+)\.(\d+)\](.*)/i )
	{
		# now - uptime + sekunden
		my $t_time = scalar localtime( $t_now + $1 );
		push( @dmesg_new , "[ $t_time ] $3" );
	}
}

print join( "\n", @dmesg_new );
print "\n";

File: dmesg.php
#!/usr/bin/php
<?php
// v0.1
error_reporting(0);
set_time_limit(0);
@ob_implicit_flush(true);
@ob_end_flush();
//$S="&nbsp;";
$S=" ";

$lines = file("/var/log/dmesg");
for ($i=0; $i<count($lines); $i++) {
    $zeile = trim($lines[$i]);
    $EndTpos1=(strpos($zeile, "]") - 1);
    $EndTpos2=(strpos($zeile, "]") + 1);
    $TIME=trim(substr($zeile, 1, $EndTpos1));
    $TIME=date("H:i:s d.m.Y", dmesgTime($TIME));
    $LINE=substr("$zeile", $EndTpos2);
    printf("$line %s", "[".$TIME."]  ".str_replace(" ", $S, $LINE)."\n");
}

function dmesgTime($time) {
    global $Uptime;
    $now = time();
    if (!isset($Uptime)) { $Uptime = shell_exec("cat /proc/uptime | cut -d'.' -f1"); }
    $t_now = $now - $Uptime;
    $t_time = $t_now + $time;
    return $t_time;
}
?>
